test/torchaudio_unittest/functional/functional_impl.py

```diff
@@ -13,6 +13,35 @@
 
 
 class Functional(TestBaseMixin):
+    def _test_resample_waveform_accuracy(self, up_scale_factor=None, down_scale_factor=None,
+                                         resampling_method="sinc_interpolation", atol=1e-1, rtol=1e-4):
+        # resample the signal and compare it to the ground truth
+        n_to_trim = 20
+        sample_rate = 1000
+        new_sample_rate = sample_rate
+
+        if up_scale_factor is not None:
+            new_sample_rate *= up_scale_factor
+
+        if down_scale_factor is not None:
+            new_sample_rate //= down_scale_factor
```

Comment:
I think, simply passing the original sample rate and the new sample rate is more readable, rather than `up_scale_factor` / `down_scale_factor`.

But since this PR is about moving the tests, so this does not need to be addressed.

Reply:
Please add this^ as a follow-up work item. The implementation can be much simplified.